src/fairseq2/recipes/lm/preference_finetune/recipe.py

```diff
@@ -140,6 +140,9 @@ class PreferenceOptimizationConfig:
     checkpoint_every_n_steps: int = 1000
     """The step interval at which to checkpoint."""
 
+    checkpoint_every_n_data_epochs: Optional[int] = None
```

Comment:
Nit: we recently moved to Python 3.10 and started using the "pipe" syntax for optionals and unions. Please change this and the public metrics fields to `: int | None = None` synax (see other fields as example.